src/components/Scrollbar/HorizontalScrollbar.jsx

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import Srollbar, {SCROLLBAR_TYPE, BUTTON_SCROLL_STEP} from './Scrollbar.jsx';
+
+import {
+	EVENT_SCROLABLE,
+	SCROLLABLE_CONTEXT_EMITTER,
+} from '../Scrollable/Scrollable.const';
+
+import {PURE} from 'dx-util/src/react/pure';
+import {themr} from 'react-css-themr';
+
+export const HORIZONTAL_SCROLLBAR = Symbol('HorizontalScrollbar');
+
+@PURE
+@themr(HORIZONTAL_SCROLLBAR)
+export default class HorizontalScrollbar extends Srollbar {
+
+	static propTypes = {
+		...Srollbar.PropTypes,
+		scrollLeft: React.PropTypes.number
+	}
+
+	_ratio;
+	_previousDragCoordinate;
+
+	/**
+	 * @returns {Number}
+	 * @protected
+	 */
+	_getMinBarSize() {
+		return parseFloat(window.getComputedStyle(this._bar).getPropertyValue('min-width'));
+	}
+
+	/**
+	 * Hide native scroll
+	 * @private
+	 */
+	_hideNativaScrollContainer() {
+		const {container} = this.props;
+		const {height} = this.context.size;
+		container.style.marginBottom = `-${height}px`;
+		container.style.height = `calc(100% + ${height}px)`;
+	}
+
+	componentDidMount() {
+		super.componentDidMount();
+		this._minBarSize = this._getMinBarSize();
+		this._hideNativaScrollContainer();
```

Comment:
Dragging behavior is inconsistent: mousedown, drag out of scrollbar, mouseup.
Why do you need the buttons? Hold is not working properly, hover/pressed state is inconsistent.